models/metadatainstancemodel.js

```diff
@@ -42,21 +42,7 @@ function MetadataInstanceModel(cfg) {
 util.inherits(MetadataInstanceModel, MetadataModel);
 
 MetadataInstanceModel.prototype.getAdminScopes = function(cb) {
-  var query = [
-    'SELECT DISTINCT ON (s.id_scope) s.id_scope as id, s.scope_name as name, s.status, s.timezone,',
-    '(CASE WHEN (s.parent_id_scope IS NULL) THEN true ELSE false END) AS multi,',
-    'array(SELECT DISTINCT c.id_category FROM metadata.categories_scopes c',
-    'WHERE s.status = 1 AND c.id_scope = s.id_scope ORDER BY c.id_category',
-    ') AS categories, ',
-    'array(SELECT json_build_object(\'name\', name, \'surname\', surname) FROM public.users ',
-    'where users_id = ANY((SELECT read_users FROM public.users_graph where name=s.id_scope )::bigint[]) GROUP BY users_id)  as users',
-    'FROM metadata.scopes s LEFT JOIN metadata.variables_scopes v on s.id_scope=v.id_scope',
-    'WHERE (s.parent_id_scope=\'orphan\' OR s.parent_id_scope is NULL)'
-  ];
-  this.query(query.join(' '), null, function(err, d) {
-    if (err) return cb(err);
-    return cb(null, d.rows);
-  });
+  this.getScopeForAdmin(null, cb);
```

Comment:
@ajaest could we see tomorrow the objective of this? It's only for understanding the problem and the solution you propose, so I can review better. By the moment you are removing two queries for using only one (and with pretty exquisite code) 😍

Reply:
I am not quite happy with the resulting naming decisions taken here. I would have liked to make the prototype of ``getScopeForAdmin`` as ``getScope**s**ForAdmin(cb, include=[])``, so that ``getAdminScopes`` and ``getScopeForAdmin`` could be deleted. That way, the routes could be written as
```javascript
router.get('/', function(req, res, next) {
   new ScopeModel().getScopesForAdmin((scopes)=>{
      res.json(scopes);
   });
});

router.get('/:scope', checkScope, function(req, res, next) {
  router.get('/', function(req, res, next) {
     new ScopeModel().getScopesForAdmin((scopes)=>{
        res.json(scopes[0]);
     }, [req.params.scope]);
  });
});
```

However, I realized that ``getAdminScopes`` and ``getScopeForAdmin`` where used here and there, and I did not wanted to go very deep into refactoring (yet).

Since the new ``getScopeForAdmin`` returns a format which cleanly merges the formats returned by ``getAdminScopes`` and ``getScopeForAdmin`, other functions using them should work as before.

Reply:
Nice!!